server/src/main/java/org/cloudfoundry/identity/uaa/provider/JdbcIdentityProviderProvisioning.java

```diff
@@ -56,6 +61,24 @@ public JdbcIdentityProviderProvisioning(JdbcTemplate jdbcTemplate) {
         this.jdbcTemplate = jdbcTemplate;
     }
 
+    public boolean idpWithAliasExistsInZone(final String zoneId) {
+        final Integer idpWithAliasExists;
+        try {
+            idpWithAliasExists = jdbcTemplate.queryForObject(
+                    IDP_WITH_ALIAS_EXISTS_QUERY,
+                    new Object[]{zoneId},
+                    new int[]{VARCHAR},
+                    Integer.class
+            );
```

Comment:
Actually, I think it would be better, more efficient if you call `jdbcTemplate.queryForList()` instead and then just check whether the list is empty or not.

Reply:
fixed with commit a09808b